joomla/joomla-cms
# Abstract nulldate

#### Summary of Changes

Each database driver has it's own nulldate, we should be checking against that rather than hardcoding a nulldate. This PR fixes the few places where the database driver nulldate is not being used yet.
#### Testing Instructions
1. Go to Users and create a new user in the backend
2. Check that the user appears in the list and the last visit date is showing Never
3. Apply the patch
4. Reload the list and the last visit date should still say Never
5. Go to Content -> Featured Articles
6. Set the unpublish date for the Beginners article to a date in the past
7. Login to the front-end as super user
8. Check the article now contains the div `<div class="system-unpublished">`
9. Apply the patch
10. Check the div `<div class="system-unpublished">` is still there
11. Login to the front-end of the website
12. Go to Your Profile (this is the user profile page)
13. Check that the Last visit date is there
14. Apply the patch
15. Check that the last visit date is still there
16. Apply the patch
17. Check that new users can activate themselves in the User Options
18. Create a new user on the front-end
19. Check that the email with the activation link works so that the user is activated
20. Check that the user is activated
21. Create a new user in the front-end
22. In the back-end activate the user as a super user/system administrator
23. Check that the user is activated
